compiler/parser/chapel.lex

```diff
@@ -468,6 +487,63 @@ static const char* eatStringLiteral(yyscan_t scanner, const char* startChar) {
   return astr(stringBuffer);
 }
 
+static const char* eatMultilineStringLiteral(yyscan_t scanner,
+                                             const char* startChar) {
+  YYLTYPE*   yyLloc  = yyget_lloc(scanner);
+  const char startCh = *startChar;
+  int startChCount = 0;
+  int        c       = 0;
+
+  newString();
+
+  while (((c = getNextYYChar(scanner)) != startCh || startChCount < 2) && c != 0) {
```

Comment:
I think this would be clearer to write as say 'while true' with a break in the event we've just read 3 of the start character. I'm finding the body of this while conditional to be a bit more complex than I'd like.